lib/ClangImporter/ClangImporter.cpp

```diff
@@ -358,6 +358,39 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
       // Just use the most feature-rich C language mode.
       "-x", "c", "-std=gnu11",
     });
+
+    // The module map used for Glibc depends on the target we're compiling for,
+    // and is not included in the resource directory with the other implicit
+    // module maps. It's at {freebsd|linux}/{arch}/glibc/module.map.
+    llvm::SmallString<128> GlibcModuleMapPath;
+    if (!importerOpts.OverrideResourceDir.empty()) {
+      GlibcModuleMapPath =
+        llvm::SmallString<128>(importerOpts.OverrideResourceDir);
```

Comment:
Should be no need to cast here. If there's not an implicit conversion, please use `.append(…)` instead. (Or try converting through StringRef instead of SmallString.)